utility/service_node.go

```diff
@@ -0,0 +1,392 @@
+package utility
+
+import (
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/types"
+	utilTypes "github.com/pokt-network/pocket/utility/types"
+)
+
+func (u *UtilityContext) HandleMessageStakeServiceNode(message *utilTypes.MessageStakeServiceNode) types.Error {
+	publicKey, er := crypto.NewPublicKeyFromBytes(message.PublicKey)
+	if er != nil {
+		return types.ErrNewPublicKeyFromBytes(er)
+	}
+	// ensure above minimum stake
+	minStake, err := u.GetServiceNodeMinimumStake()
+	if err != nil {
+		return err
+	}
+	amount, err := types.StringToBigInt(message.Amount)
+	if err != nil {
+		return err
+	}
+	if types.BigIntLessThan(amount, minStake) {
+		return types.ErrMinimumStake()
+	}
+	// ensure signer has sufficient funding for the stake
+	signerAccountAmount, err := u.GetAccountAmount(message.Signer)
+	if err != nil {
+		return err
+	}
+	signerAccountAmount.Sub(signerAccountAmount, amount)
+	if signerAccountAmount.Sign() == -1 {
+		return types.ErrInsufficientAmountError()
+	}
+	maxChains, err := u.GetServiceNodeMaxChains()
+	if err != nil {
+		return err
+	}
+	// validate chains
+	if len(message.Chains) > maxChains {
+		return types.ErrMaxChains(maxChains)
+	}
```

Comment:
Agreed

Reply:
Oops on second thought, disagree:

Validate Chains is needed here because it's stateful

ValidateBasic is not stateful.

ValidateRelayChains is a stateless function for ValidateBasic